climakitae/tmy.py

```diff
@@ -1,194 +1,346 @@
+            Applicable for both the historical and future periods.
+            Returns: list of tmy
+            """
+            hourly_list = []
+            for x in np.arange(1, days_in_year+1, 1):
+                data_on_day_x = data.where(data.time.dt.dayofyear == x, drop = True)
+                data_grouped = data_on_day_x.groupby("time.hour")
+                mean_by_hour = data_grouped.mean()
+                min_diff = abs(data_grouped - mean_by_hour).groupby("time.hour").min()
+                typical_hourly_data_on_day_x = data_on_day_x.where(abs(data_grouped - mean_by_hour).groupby("time.hour") == min_diff, drop = True).sortby("time.hour")
+                np_typical_hourly_data_on_day_x = remove_repeats(typical_hourly_data_on_day_x)
+                hourly_list.append(np_typical_hourly_data_on_day_x)
+            return hourly_list
+
+        tmy_hist = tmy_calc(self.historical_tmy_data, days_in_year=days_in_year)
+        tmy_future = tmy_calc(self.future_tmy_data, days_in_year=days_in_year)
+
+
+        ## Funnel data into pandas DataFrame object
+        df_hist = pd.DataFrame(tmy_hist, columns=np.arange(1,25,1), index=np.arange(1,days_in_year+1,1))
+        df_hist = df_hist.iloc[::-1]
+        df_future = pd.DataFrame(tmy_future, columns = np.arange(1,25,1), index=np.arange(1,days_in_year+1,1))
+        df_future = df_future.iloc[::-1]
+
+        # update heatmap df and title with selections
+        if self.tmy_options == "Absolute":
+            if self.abs_tmy_options == "Historical":
+                df = df_hist
+                title = "Average Meteorological Year\nAbsolute {} Baseline \n{}".format(self.abs_tmy_options, self.cached_area2)
+                clabel = self.variable2 + " (" +self.historical_tmy_data.attrs["units"]+")"
+            else:
+                df = df_future
+                title = "Average Meteorological Year\nAbsolute {} at {}°C \n{}".format(self.abs_tmy_options, self.warmlevel, self.cached_area2)
+                clabel = self.variable2 + " (" +self.future_tmy_data.attrs["units"]+")"
+        elif self.tmy_options == "Difference":
+            if self.diff_tmy_options == "Warming Level Future":
+                df = df_future - df_hist
+                title = "Average Meteorological Year\nDifference between {} at {}°C and Historical Baseline \n{}".format(self.diff_tmy_options, self.warmlevel, self.cached_area2)
+                clabel = self.variable2 + " (" +self.historical_tmy_data.attrs["units"]+")"
+            else:
+                df = df_future - df_hist # placeholder for now for severe amy
+                title = "Average Meteorological Year\nDifference between {} at 90th percentile and Historical Baseline \n{}".format(self.diff_tmy_options, self.cached_area2)
+                clabel = self.variable2 + " (" +self.historical_tmy_data.attrs["units"]+")"
+        else:
+            title = "Average Meteorological Year\n{}".format(self.cached_area2)
```

Comment:
I'm not sure if I like the stacking title set up for the heatmap. The title is a strange width (quite narrow) compared to the width of the plot. I would suggest having a two-line title instead. Perhaps something like this for a few example selections: 

- Average Meteorological Year: Fresno County
  Data: Historical Baseline
- Average Meteorological Year: California 
 Data: 3deg Warming Level Future minus Historical Baseline
- Average Meteorological Year: Santa Clara Watershed 
 Data: 1.5deg Warming Level Future

Although I don't know if I like this that much either.... bleh. I would just go with your gut. Above is just a suggestion.

Reply:
@elehmer do you have any opinions about the title of the heatmap? Perhaps if we all put our heads together we can think of a good solution. 3 lines of stacking title feels a bit strange to me, but maybe thats the clearest method.

Reply:
Happy to hear any suggestions on this! One of the things I noted from the WG feedback, and my own observations, is that it can be pretty easy to get lost in the weeds about what is being displayed and for what location, which is why I thought putting the location in the title to make it obvious! The nice thing is that when you download the plot as a figure, it does keep that title with location so there's not any confusion about what is **exactly** being displayed.